src/main/java/com/timgroup/statsd/CgroupReader.java

```diff
@@ -70,4 +116,84 @@ public static String parse(final String cgroupsContent) {
 
         return null;
     }
+
+    /**
+     * Returns true if the host cgroup namespace is used.
+     * It looks at the inode of `/proc/self/ns/cgroup` and compares it to
+     * HOST_CGROUP_NAMESPACE_INODE.
+     * 
+     * @param path Path to the cgroup namespace file.
+     */
+    private static boolean isHostCgroupNamespace(final Path path) {
+        long hostCgroupInode = inodeForPath(path);
+        return hostCgroupInode == HOST_CGROUP_NAMESPACE_INODE;
+    }
+
+    /**
+     * Returns the inode for the given path.
+     * 
+     * @param path Path to the cgroup namespace file.
+     */
+    private static long inodeForPath(final Path path) {
+        try {
+            long inode = (long) Files.getAttribute(path, "unix:ino");
+            return inode;
+        } catch (Exception e) {
+            return 0;
+        }
+    }
+
+    /**
+     * Returns the cgroup controller inode for the given cgroup mount path and
+     * procSelfCgroupPath.
+     * 
+     * @param cgroupMountPath Path to the cgroup mount point.
+     * @param cgroupContent   String content of the cgroup file.
+     */
+    public static String getCgroupInode(final Path cgroupMountPath, final String cgroupContent) throws IOException {
+        Map<String, String> cgroupControllersPaths = parseCgroupNodePath(cgroupContent);
+        if (cgroupControllersPaths == null) {
+            return null;
+        }
+
+        // Retrieve the cgroup inode from /sys/fs/cgroup+controller+cgroupNodePath
+        List<String> controllers = Arrays.asList(CGROUPV1_BASE_CONTROLLER, CGROUPV2_BASE_CONTROLLER);
+        for (String controller : controllers) {
+            String cgroupNodePath = cgroupControllersPaths.get(controller);
+            if (cgroupNodePath == null) {
+                continue;
+            }
+            Path path = Paths.get(cgroupMountPath.toString(), controller, cgroupNodePath);
+            long inode = inodeForPath(path);
+            if (inode > 2) {
```

Comment:
Could you please add a comment about this condition? Why 2?

Reply:
Added a comment. Basically inode <= 2 is an error case or can be a case where the app is running on host.